flink-ml-benchmark/src/main/java/org/apache/flink/ml/benchmark/datagenerator/common/LabeledPointWithWeightGenerator.java

```diff
@@ -0,0 +1,202 @@
+    }
+
+    public LabeledPointWithWeightGenerator setWeightArity(int value) {
+        return set(WEIGHT_ARITY, value);
+    }
+
+    public LabeledPointWithWeightGenerator() {
+        ParamUtils.initializeMapWithDefaultValues(paramMap, this);
+    }
+
+    @Override
+    public Table[] getData(StreamTableEnvironment tEnv) {
+        StreamExecutionEnvironment env = TableUtils.getExecutionEnvironment(tEnv);
+
+        DataStream<Row> dataStream =
+                env.fromParallelCollection(
+                                new NumberSequenceIterator(1L, getNumValues()),
+                                BasicTypeInfo.LONG_TYPE_INFO)
+                        .map(
+                                new RandomLabeledPointWithWeightGenerator(
+                                        getSeed(),
+                                        getVectorDim(),
+                                        getFeaturesArity(),
+                                        getLabelArity(),
+                                        getWeightArity()),
+                                new RowTypeInfo(
+                                        new TypeInformation[] {
+                                            DenseVectorTypeInfo.INSTANCE, Types.DOUBLE, Types.DOUBLE
+                                        },
+                                        new String[] {
+                                            getFeaturesCol(), getLabelCol(), getWeightCol()
+                                        }));
+
+        Table dataTable = tEnv.fromDataStream(dataStream);
+
+        return new Table[] {dataTable};
+    }
+
+    @Override
+    public Map<Param<?>, Object> getParamMap() {
+        return paramMap;
+    }
+
+    private static class RandomLabeledPointWithWeightGenerator extends RichMapFunction<Long, Row> {
+        private final long initSeed;
+        private final int vectorDim;
+        private final int featuresArity;
+        private final int labelArity;
+        private final int weightArity;
+        private Random random;
+
+        private RandomLabeledPointWithWeightGenerator(
+                long initSeed, int vectorDim, int featuresArity, int labelArity, int weightArity) {
+            this.initSeed = initSeed;
+            this.vectorDim = vectorDim;
+            this.featuresArity = featuresArity;
+            this.labelArity = labelArity;
+            this.weightArity = weightArity;
+        }
+
+        @Override
+        public void open(Configuration parameters) throws Exception {
+            super.open(parameters);
+            int index = getRuntimeContext().getIndexOfThisSubtask();
+            random = new Random(Tuple2.of(initSeed, index).hashCode());
+        }
+
+        @Override
+        public Row map(Long aLong) {
+            double[] values = new double[vectorDim];
+            if (featuresArity > 0) {
+                for (int i = 0; i < vectorDim; i++) {
+                    values[i] = random.nextInt(featuresArity);
```

Comment:
It is probably simpler to add the following method instead of repeating this code for values, label, and weight.

```
private double getValue(int arity, Random random) {
    if (arity > 0) {
        return random.nextInt(arity);
    }
    return random.nextDouble();
}
```